units/periodic_builds.go

```diff
@@ -96,25 +96,22 @@ func (j *periodicBuildJob) Run(ctx context.Context) {
 		}))
 	}()
 
-	mostRecentVersion, err := model.FindVersionByLastKnownGoodConfig(j.ProjectID, -1)
+	mostRecentRevision, err := model.FindLatestRevisionForProject(j.ProjectID)
```

Comment:
nit: Should we error if mostRecentRevision = "" (since we expect it to always have one)?

Reply:
This happens in FindLatestRevisionForProject